src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -0,0 +1,269 @@
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <stdint.h>
+#include <set>
+#include <memory>
+
+#include "gtest/gtest.h"
+#include "utils/push_log.h"
+#include "utils/lock.h"
+#include "utils/file_system.h"
+#include "utils/make_shared.h"
+#include "utils/custom_string.h"
+#include "policy/mock_policy_settings.h"
+#include "application_manager/application.h"
+#include "application_manager/application_impl.h"
+#include "application_manager/application_manager_impl.h"
+#include "application_manager/mock_resumption_data.h"
+#include "application_manager/resumption/resume_ctrl_impl.h"
+#include "application_manager/mock_application_manager_settings.h"
+#include "hmi_message_handler/mock_hmi_message_handler.h"
+#include "connection_handler/mock_connection_handler.h"
+#include "protocol_handler/mock_session_observer.h"
+#include "policy/usage_statistics/mock_statistics_manager.h"
+#include "application_manager/test/include/application_manager/mock_message_helper.h"
+
+namespace test {
+namespace components {
+namespace application_manager_test {
+
+namespace am = application_manager;
+namespace policy_test = test::components::policy_handler_test;
+namespace con_test = connection_handler_test;
+
+using testing::_;
+using ::testing::Mock;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::NiceMock;
+using ::testing::SetArgReferee;
+using ::testing::SetArgPointee;
+using ::testing::Truly;
+using ::testing::AtLeast;
+using ::testing::DoAll;
+using usage_statistics_test::MockStatisticsManager;
+using namespace application_manager;
+
+namespace {
+const std::string kHmiCapabilitiesFile("hmi_capabilities.json");
+const uint32_t kThreadPoolSize = 1;
+const bool kHmiLaunch = false;
+const uint32_t kStopStreamingTimeout = 10000;
+const uint32_t kWaitFreeQueue = 2000;
+const std::string kDirectoryName = "./test_storage";
+const uint32_t kTimeout = 10000u;
+const std::string kFolderName = "app";
+// const std::string kTestPath =
+//    file_system::ConcatPath(kDirectoryName, kFolderName);
+// const std::string kTestFileName =
+//    file_system::ConcatPath(kTestPath, "test_file2.bmp");
+const std::string kMacAddr_("kMacAddr_ess");
+
+bool is_called_ = false;
+uint32_t hmi_app_id_ = 0u;
+sync_primitives::Lock state_lock_;
+sync_primitives::ConditionalVariable state_condition_;
+}  // namespace
+
+class ApplicationManagerImplTest : public ::testing::Test {
+ public:
+  ApplicationManagerImplTest()
+      : mock_message_helper_(
+            application_manager::MockMessageHelper::message_helper_mock()) {}
+
+ protected:
+  void SetUp() OVERRIDE {
+    app_id_ = 10;
```

Comment:
@LuxoftAKutsan done in https://github.com/smartdevicelink/sdl_core/pull/1234/commits/817018f15bcc9a280711589e063f3a1a262e115b